Sources/PrimerSDK/Classes/PCI/User Interface/Text Fields/PrimerCardNumberFieldView.swift

```diff
@@ -87,6 +87,23 @@ public final class PrimerCardNumberFieldView: PrimerTextFieldView {
         }
 
         primerTextField.text = newText.withoutWhiteSpace.separate(every: 4, with: " ")
+
+        // Detect pasting of the card number
+        if string.count > 1 && (primerTextField.internalText?.count ?? 0) > 1 {
+            if let text = primerTextField.internalText {
+
+                // Get the position of the last character in the string
+                if let endPosition = primerTextField.position(from: primerTextField.beginningOfDocument, offset: text.count) {
+                    // Delaying it for 0.01 seconds because setting the selectedTextRange
+                    // before `shouldChangeCharactersIn:range` returns does not apply the value.
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.01, execute: {
```

Comment:
If this is already on main queue (which it should be) can you just dispatch to main without the delay? That should make it happen as soon as this function exits

Reply:
true, thanks!